File: ch8/yojkim/ch8/franc.go
package ch8

type Franc struct {
	amount int
}

func NewFranc(amount int) Franc {
	return Franc{amount}
}

func (f Franc) Times(multiplier int) Money {
	return Franc{f.amount * multiplier}
}

func (f Franc) Equal(input Money) bool {
	franc, ok := input.(Franc)
	if !ok {
		return false
	}
	return f.amount == franc.amount
}

File: ch6/gunddol/src/ch6/Franc.java
package ch6;

public class Franc extends Money{

    public Franc(int amount){
        this.amount = amount;
    }

    public Franc times(int multiplier) {
        return new Franc(amount * multiplier);
    }

    public boolean equals(Object object){
        Money money = (Money) object;
        return amount == money.amount;
    }
}
